doc/charm++/manual.rst

```diff
@@ -10976,6 +10976,34 @@ superclass to do the final delivery after you’ve sent your messages.
 Experimental Features
 =====================
 
+CMI-SHMEM
+---------
+CMI-SHMEM is an experimental IPC module for Charm++, with the goal of
+adding fast IPC to any machine layer. Each process opens a shared segment
+containing an MPSC queue and a memory pool. Processes can allocate "blocks"
+of memory from their peers pools, and "push" them onto their MPSC queue.
+This is mechanically similar to the PXSHM layer, but "address-free" atomics
+are used for synchronization instead of memory fences, etc. The C++ recommends
+that its atomics are address-free, i.e., ordering is enforced no matter which
+virtual address is used to access a physical address; however, this is not
+guaranteed. In practice, all modern compilers comply with this recommendation,
+but please alert us if you encounter synchronization issues!
+
+To build with CMI-SHMEM, pass ``--enable-shmem`` (to use PXSHM) or
+``--enable-xpmem`` (to use XPMEM) as command-line options to a Charm++ build;
+only CMake builds are supported at this time. By default, Charm++ will only
+use CMI-SHMEM for "small" messages, after which it is recommended to use other
```

Comment:
I would say Charm++ uses CMI-Shmem for Charm++ messages (which can be any size), but not for zero copy API put/get operations. The zero copy API and Charm++ messages are both in theory general purpose communication mechanisms, and the metadata message of a zero copy API transfer does use a Charm++ message, so there are some complicating matters.

Reply:
Would you say that in addition to what is here? My intent was to convey that CMI-SHMEM has a threshold, i.e.:
`len > (CmiRecommendedIpcBlockCutoff() + sizeof(envelope))`

Reply:
Yeah, hard to make it any clearer without making it overly complicated. It's fine as is.